### PR TITLE
Ticket Assignment Update Note

### LINKED ISSUE #959: The contribution docs should make it clear that the contributor should explicitly claim the ticket
There was recently a case where a contributor expressed interest in a ticket by asking questions for clarification, and it was pretty obvious that they wanted to work on the ticket, but I didn't assign them to the ticket since they didn't claim it. Soon afterwards, someone else submitted a PR and simultaneously claimed the ticket, without asking if they can work on it.

There are two important pieces here:

- The contribution docs should state that the ticket should be claimed if you have desire to work on the ticket.
- We should be proactive about asking to assign the ticket to the contributor if they have expressed interest, or 0/5 even assign them preemptively before they explicitly claim the ticket.

### PR DESCRIPTION
**Pull request by Rutam21**
## Summary

This PR adds an extra note for Ticket assignment to contributors.

## Ticket Link

Fixes #959

### CODE REVIEW

**Comment by Rutam21** [comment]
@cwarnermm Please review and suggest changes, if any. Thanks.

File: site/content/contribute/more-info/getting-started/contribution-checklist.md [hunk 1]
```diff
@@ -16,22 +16,24 @@ Follow this checklist for submitting a pull request (PR):
 
 1. You've signed the {{< newtabref href="https://mattermost.com/mattermost-contributor-agreement/" title="Contributor License Agreement" >}}, so you can be added to the Mattermost {{< newtabref href="https://docs.google.com/spreadsheets/d/1NTCeG-iL_VS9bFqtmHSfwETo5f-8MQ7oMDE5IUYJi_Y/pubhtml?gid=0&single=true" title="Approved Contributor List" >}}.
     - If you've included your mailing address in the signed {{< newtabref href="https://mattermost.com/mattermost-contributor-agreement/" title="Contributor License Agreement" >}}, you may receive a {{< newtabref href="https://forum.mattermost.com/t/limited-edition-mattermost-mugs/143" title="Limited Edition Mattermost Mug" >}} as a thank you gift after your first pull request is merged.
-2. Your ticket is a Help Wanted GitHub issue for the Mattermost project you're contributing to.
+2. You have claimed the ticket that you wish to work on by asking for an assignment from the Mattermost team.
+   - Tickets are assigned on a first-come-first-serve basis.
+4. Your ticket is a Help Wanted GitHub issue for the Mattermost project you're contributing to.
```
Comment (cwarnermm) [inline site/content/contribute/more-info/getting-started/contribution-checklist.md:21]: @Rutam21 - I think the step numbering is off by one from this point to the end.

Comment (hanzei) [reply site/content/contribute/more-info/getting-started/contribution-checklist.md:21]: Blocking on this

**Review by hanzei — CHANGES_REQUESTED** [review]
Besides the numbering issue, LGTM :+1:

File: site/content/contribute/more-info/getting-started/contribution-checklist.md (hunk 1 above)
Comment (cwarnermm) [inline site/content/contribute/more-info/getting-started/contribution-checklist.md:21]:
```suggestion
3. Your ticket is a Help Wanted GitHub issue for the Mattermost project you're contributing to.
```

File: site/content/contribute/more-info/getting-started/contribution-checklist.md [hunk 2]
```diff
@@ -16,22 +16,24 @@ Follow this checklist for submitting a pull request (PR):
...
-2. Your ticket is a Help Wanted GitHub issue for the Mattermost project you're contributing to.
+2. You have claimed the ticket that you wish to work on by asking for an assignment from the Mattermost team.
+   - Tickets are assigned on a first-come-first-serve basis.
+3. Your ticket is a Help Wanted GitHub issue for the Mattermost project you're contributing to.
     - If not, follow the process [here]({{< ref "/contribute/more-info/getting-started/contributions-without-ticket" >}}).
-3. Your code is thoroughly tested, including appropriate [unit, end-to-end, and integration tests for webapp]({{< ref "/contribute/more-info/getting-started/test-guideline" >}}).
-4. If applicable, user interface strings are included in localization files:
+5. Your code is thoroughly tested, including appropriate [unit, end-to-end, and integration tests for webapp]({{< ref "/contribute/more-info/getting-started/test-guideline" >}}).
```
Comment (cwarnermm) [inline site/content/contribute/more-info/getting-started/contribution-checklist.md:23]:
```suggestion
4. Your code is thoroughly tested, including appropriate [unit, end-to-end, and integration tests for webapp]({{< ref "/contribute/more-info/getting-started/test-guideline" >}}).
```

File: site/content/contribute/more-info/getting-started/contribution-checklist.md [hunk 3]
```diff
@@ -16,22 +16,24 @@ Follow this checklist for submitting a pull request (PR):
...
-2. Your ticket is a Help Wanted GitHub issue for the Mattermost project you're contributing to.
+2. You have claimed the ticket that you wish to work on by asking for an assignment from the Mattermost team.
+   - Tickets are assigned on a first-come-first-serve basis.
+3. Your ticket is a Help Wanted GitHub issue for the Mattermost project you're contributing to.
     - If not, follow the process [here]({{< ref "/contribute/more-info/getting-started/contributions-without-ticket" >}}).
-3. Your code is thoroughly tested, including appropriate [unit, end-to-end, and integration tests for webapp]({{< ref "/contribute/more-info/getting-started/test-guideline" >}}).
-4. If applicable, user interface strings are included in localization files:
+4. Your code is thoroughly tested, including appropriate [unit, end-to-end, and integration tests for webapp]({{< ref "/contribute/more-info/getting-started/test-guideline" >}}).
+6. If applicable, user interface strings are included in localization files:
```
Comment (cwarnermm) [inline site/content/contribute/more-info/getting-started/contribution-checklist.md:24]:
```suggestion
5. If applicable, user interface strings are included in localization files:
```

File: site/content/contribute/more-info/getting-started/contribution-checklist.md [hunk 4]
```diff
@@ -16,22 +16,24 @@ Follow this checklist for submitting a pull request (PR):
...
-7. If your PR adds or changes a RESTful API endpoint, please update the {{< newtabref href="https://github.com/mattermost/mattermost-api-reference" title="API documentation" >}}.
-8. If your PR adds a new plugin API method or hook, please add an example to the {{< newtabref href="https://github.com/mattermost/mattermost-plugin-starter-template" title="Plugin Starter Template" >}}.
-9. If QA review is applicable, your PR includes test steps or expected results.
-10. If the PR adds a substantial feature, a feature flag is included. Please see [criteria here]({{< ref "/contribute/more-info/server/feature-flags#when-to-use" >}}).
-11. Your PR includes basic documentation about the change/addition you're submitting. View our {{< newtabref href="https://handbook.mattermost.com/operations/research-and-development/product/technical-writing-team-handbook#submit-documentation-with-your-pr-community" title="guidelines" >}} for more information about submitting documentation and the review process.
+    5.1. In the webapp repository run `make i18n-extract` to generate the new/updated strings.
+7. The PR is submitted against the Mattermost `master` branch from your fork.
```
Comment (cwarnermm) [inline site/content/contribute/more-info/getting-started/contribution-checklist.md:30]:
```suggestion
6. The PR is submitted against the Mattermost `master` branch from your fork.
```

File: site/content/contribute/more-info/getting-started/contribution-checklist.md [hunk 5]
```diff
@@ -16,22 +16,24 @@ Follow this checklist for submitting a pull request (PR):
...
+8. The PR title begins with the Jira or GitHub ticket ID (e.g. `[MM-394]` or `[GH-394]`) and summary template is filled out.
```
Comment (cwarnermm) [inline site/content/contribute/more-info/getting-started/contribution-checklist.md:31]:
```suggestion
7. The PR title begins with the Jira or GitHub ticket ID (e.g. `[MM-394]` or `[GH-394]`) and summary template is filled out.
```

File: site/content/contribute/more-info/getting-started/contribution-checklist.md [hunk 6]
```diff
@@ -16,22 +16,24 @@ Follow this checklist for submitting a pull request (PR):
...
+9. If your PR adds or changes a RESTful API endpoint, please update the {{< newtabref href="https://github.com/mattermost/mattermost-api-reference" title="API documentation" >}}.
```
Comment (cwarnermm) [inline site/content/contribute/more-info/getting-started/contribution-checklist.md:32]:
```suggestion
8. If your PR adds or changes a RESTful API endpoint, please update the {{< newtabref href="https://github.com/mattermost/mattermost-api-reference" title="API documentation" >}}.
```

File: site/content/contribute/more-info/getting-started/contribution-checklist.md [hunk 7]
```diff
@@ -16,22 +16,24 @@ Follow this checklist for submitting a pull request (PR):
...
+10. If your PR adds a new plugin API method or hook, please add an example to the {{< newtabref href="https://github.com/mattermost/mattermost-plugin-starter-template" title="Plugin Starter Template" >}}.
```
Comment (cwarnermm) [inline site/content/contribute/more-info/getting-started/contribution-checklist.md:33]:
```suggestion
9. If your PR adds a new plugin API method or hook, please add an example to the {{< newtabref href="https://github.com/mattermost/mattermost-plugin-starter-template" title="Plugin Starter Template" >}}.
```

File: site/content/contribute/more-info/getting-started/contribution-checklist.md [hunk 8]
```diff
@@ -16,22 +16,24 @@ Follow this checklist for submitting a pull request (PR):
...
+11. If QA review is applicable, your PR includes test steps or expected results.
```
Comment (cwarnermm) [inline site/content/contribute/more-info/getting-started/contribution-checklist.md:34]:
```suggestion
10. If QA review is applicable, your PR includes test steps or expected results.
```

File: site/content/contribute/more-info/getting-started/contribution-checklist.md [hunk 9]
```diff
@@ -16,22 +16,24 @@ Follow this checklist for submitting a pull request (PR):
...
+12. If the PR adds a substantial feature, a feature flag is included. Please see [criteria here]({{< ref "/contribute/more-info/server/feature-flags#when-to-use" >}}).
```
Comment (cwarnermm) [inline site/content/contribute/more-info/getting-started/contribution-checklist.md:35]:
```suggestion
11. If the PR adds a substantial feature, a feature flag is included. Please see [criteria here]({{< ref "/contribute/more-info/server/feature-flags#when-to-use" >}}).
```

File: site/content/contribute/more-info/getting-started/contribution-checklist.md [hunk 10]
```diff
@@ -16,22 +16,24 @@ Follow this checklist for submitting a pull request (PR):
...
+13. Your PR includes basic documentation about the change/addition you're submitting. View our {{< newtabref href="https://handbook.mattermost.com/operations/research-and-development/product/technical-writing-team-handbook#submit-documentation-with-your-pr-community" title="guidelines" >}} for more information about submitting documentation and the review process.
```
Comment (cwarnermm) [inline site/content/contribute/more-info/getting-started/contribution-checklist.md:36]:
```suggestion
12. Your PR includes basic documentation about the change/addition you're submitting. View our {{< newtabref href="https://handbook.mattermost.com/operations/research-and-development/product/technical-writing-team-handbook#submit-documentation-with-your-pr-community" title="guidelines" >}} for more information about submitting documentation and the review process.
```

**Review by cwarnermm — APPROVED** [review]
@hanzei - I've updated the procedural numbering. This PR is ready for your approval.

**Review by hanzei — APPROVED** [review]

**Comment by hanzei** [comment]
Thanks @Rutam21 :tada:

**Pull request merged**